### PR TITLE
Connection direction should be an optional enum

### PR DESCRIPTION
**Pull request by dbraquart**

### CODE REVIEW

File: src/components/dialogs/line-creation-dialog.js [hunk 1]
```diff
@@ -277,9 +278,11 @@ const LineCreationDialog = ({
                 editData ? true : false,
                 editData ? editData.uuid : undefined,
                 connectivity1?.connectionName?.id ?? null,
-                connectivity1?.connectionDirection?.id ?? 'UNDEFINED',
+                connectivity1?.connectionDirection?.id ??
```
Comment (Walid-Sahnoun) [inline src/components/dialogs/line-creation-dialog.js:281]: rename to connectivityOne maybe better

Comment (dbraquart) [reply src/components/dialogs/line-creation-dialog.js:281]: I've just replaced the string 'UNDEFINED' by a constant, so I will not rename connectivity1, connectivity2.
And also shuntConductance1, shuntConductance2, shuntSusceptance1, shuntSusceptance2, ...

File: src/components/dialogs/line-creation-dialog.js [hunk 2]
```diff
@@ -277,9 +278,11 @@ const LineCreationDialog = ({
...
+                    UNDEFINED_CONNECTION_DIRECTION,
                 connectivity2?.connectionName?.id ?? null,
-                connectivity2?.connectionDirection?.id ?? 'UNDEFINED'
+                connectivity2?.connectionDirection?.id ??
```
[the remark this answers is not in the document]
Comment (dbraquart) [reply src/components/dialogs/line-creation-dialog.js:284]: again we should not change these names in the whole source

[the remark this answers is not in the document]
Comment (dbraquart) [reply src/components/dialogs/line-creation-dialog.js:284]: again we should not change these names in the whole source

**Review by Walid-Sahnoun — APPROVED** [review]

**Review by bendaamerahm — APPROVED** [review]

**Pull request merged**